cpp/ql/test/experimental/query-tests/Security/CWE/CWE-200/semmle/tests/test.cpp

```diff
@@ -0,0 +1,17 @@
+typedef int FILE;
+FILE *fopen(const char *filename, const char *mode);
+int umask(int pmode);
+int chmod(char * filename,int pmode);
+int fprintf(FILE *fp,const char *fmt, ...);
+int fclose(FILE *stream);
+
+int main(int argc, char *argv[])
+{
+  //umask(0022);
+  FILE *fp;
+  fp = fopen("myFile.txt","w"); // BAD
+  //chmod("myFile.txt",0644);
+  fprintf(fp,"%s\n","data to file");
+  fclose(fp);
+  return 0;
+}
```

Comment:
It would be nice to have tests with the `umask` and `chmod` calls in place, though I appreciate they will need to be in entirely separate tests (in separate subdirectories perhaps?) for the query to work as intended.  It would also be nice to have a test where the file is read instead of written, so we can see that part of the query is working correctly.

Reply:
I will work on this shortly.

Reply:
Good afternoon.

I added new test files.
could you please run checks and show their results.

Reply:
Thanks.  It appears `test.cpp` and `test1.cpp` are identical - did you mean to delete `test.cpp` perhaps?

Reply:
i fixed `test1.cpp`